src/openrct2/entity/MoneyEffect.cpp

```diff
@@ -105,6 +91,28 @@ void MoneyEffect::Create(money64 value, const CoordsXYZ& loc)
     CreateAt(-value, offsetLoc, false);
 }
 
+/**
+ * Set the value of the money effect
+ */
+void MoneyEffect::SetValue(money64 value)
+{
+    Value = value;
+    SpriteData.Width = 64;
+    SpriteData.HeightMin = 20;
+    SpriteData.HeightMax = 30;
```

Comment:
This has to be only set once, remove that.

Reply:
What about the wiggle, does it need to be reset?

Reply:
I tried removing it, but it seems it is actually needed, as without it the bounding box glitching occurs again.

Reply:
That will reset the wave position, I think its best to have the value not do anything else, could be a different accessor if you want to influence the wiggle.

Reply:
> I tried removing it and placing it back in the createat function, but it seems it is actually needed, as without it the bounding box glitching occurs again.

I see why, apparently it doesn't use the correct method to create entities so all the default values aren't set. This is isn't great.

Reply:
Where is that?

Reply:
MoneyEffect.cpp

Reply:
Yeah, I know where the create-at function is, but I meant, where is the part where it's supposed to call that? From what I can see, in ScMap createEntityType is a template function that should call it, but I don't see it calling create functions or anything like that for an of the other entity types either.

Reply:
> Yeah, I know where the create-at function is, but I meant, where is the part where it's supposed to call that? From what I can see, in ScMap createEntityType is a template function that should call it, but I don't see it calling create functions or anything like that for an of the other entity types either.

That is the problem, the way it creates entities is wrong.

Reply:
But that's a problem far bigger than the scope of adding these two functions, no?
I wouldn't really know how to elegantly fix that.

Reply:
Yes that is a bigger problem unrelated to this PR, I guess you can leave it in for now but also reset wiggle and movement variables otherwise they have garbage data.

Reply:
All well, now?